addons/account/controllers/terms.py

```diff
@@ -4,10 +4,15 @@
 from odoo import http, _
 from odoo.http import request
 
+def sitemap_terms(env, rule, qs):
+    use_invoice_terms = env['ir.config_parameter'].sudo().get_param('account.use_invoice_terms')
+    if (use_invoice_terms and env.company.terms_type == 'html'):
+        yield {'loc': '/terms'}
```

Comment:
This is missing this part:
```suggestion
        if not qs or qs.lower() in '/terms'::
            yield {'loc': '/terms'}
```

The function passed as a `sitemap` is actually also used for page/url searches, mainly when handling links.
Both the link autocomplete search and the sitemap calls `_enumerate_pages()`, which takes a `query_string` as argument, which is what the used typed.

Without this, the `/terms` URL would be returned by the `_enumerate_pages()` regardless of the given `query_string`.